drivers/flash/spi_nor.c

```diff
@@ -969,10 +969,10 @@ static int setup_pages_layout(const struct device *dev)
 
 	uint32_t erase_size = BIT(exp);
 
-	/* Error if layout page size is not a multiple of smallest
-	 * erase size.
+	/* Error if erase size is not a multiple of
+	 * layout page size.
 	 */
-	if ((layout_page_size % erase_size) != 0) {
+	if ((erase_size % layout_page_size) != 0) {
```

Comment:
how this is related to:
838: data->page_size = jesd216_bfp_page_size(php, bfp);

Reply:
page size <= erase size